File: app/src/main/java/me/tysheng/xishi/utils/StySubscriber.java
package me.tysheng.xishi.utils;

import io.reactivex.subscribers.ResourceSubscriber;

/**
 * Created by Sty
 * Date: 16/9/16 09:09.
 */
public abstract class StySubscriber<T> extends ResourceSubscriber<T> {

    @Override
    public void onComplete() {

    }

    @Override
    public void onError(Throwable e) {

    }

    @Override
    public void onNext(T t) {
        next(t);
    }

    public abstract void next(T t);
}

File: app/src/main/java/me/tysheng/xishi/utils/StySubscriber.kt
package me.tysheng.xishi.utils

import io.reactivex.subscribers.ResourceSubscriber

/**
 * Created by Sty
 * Date: 16/9/16 09:09.
 */
abstract class StySubscriber<T> : ResourceSubscriber<T>() {

    override fun onComplete() {

    }

    override fun onError(e: Throwable) {

    }

    override fun onNext(t: T) {
        next(t)
    }

    abstract fun next(t: T)
}
